### PR TITLE
Flatten NSState and move to external parameter management

### PR DESCRIPTION
**Pull request by williamjameshandley**
This PR further simplifies the NS architecture by **eliminating state nesting** and moving inner kernel parameter management to the user's outer loop.

## Major Simplifications

**141 net lines removed**:
- ✅ **Entire `adaptive.py` module deleted** (145 lines)
- ✅ **NSState completely flattened** - no more StateWithLogLikelihood wrapper
- ✅ **inner_kernel_params removed from state** - managed externally

### 1. Flattened NSState

**Before** (nested structure):
```python
class StateWithLogLikelihood(NamedTuple):
    position: ArrayLikeTree
    logdensity: Array
    loglikelihood: Array
    loglikelihood_birth: Array

class NSState(NamedTuple):
    particles: StateWithLogLikelihood
    inner_kernel_params: Dict
```

**After** (flat structure):
```python
class NSState(NamedTuple):
    position: ArrayLikeTree
    logdensity: Array
    loglikelihood: Array
    loglikelihood_birth: Array
```

### 2. External Parameter Management

Inner kernel parameters (like covariance matrices) are now:
- Initialized externally with `init_inner_kernel_params(state)`
- Updated in the user's loop with `update_inner_kernel_params(state, info, params)`
- Passed as arguments to `step(rng_key, state, inner_kernel_params)`

This follows the SMC pattern where `lmbda` is passed separately.

## Updated Interface

```python
import jax
import jax.numpy as jnp
import blackjax
from blackjax.ns.utils import finalise
from blackjax.ns.integrator import init_integrator, update_integrator
from blackjax.ns.nss import init_inner_kernel_params, update_inner_kernel_params
import tqdm

rng_key = jax.random.PRNGKey(0)

loglikelihood_fn = lambda x: jax.scipy.stats.multivariate_normal.logpdf(
    x, jnp.ones(5), jnp.eye(5)*0.01
)
logprior_fn = lambda x: jax.scipy.stats.multivariate_normal.logpdf(
    x, jnp.zeros(5), jnp.eye(5)
)

algo = blackjax.nss(
    logprior_fn=logprior_fn,
    loglikelihood_fn=loglikelihood_fn,
    num_delete=50,
    num_inner_steps=20,
)

rng_key, initialization_key = jax.random.split(rng_key)
live = algo.init(jax.random.normal(initialization_key, (1000, 5)))
integrator = init_integrator(live)
inner_kernel_params = init_inner_kernel_params(live)
step = jax.jit(algo.step)

dead = []

with tqdm.tqdm(desc="Dead points", unit=" dead points") as pbar:
    while not (integrator.logZ_live - integrator.logZ < -3):
        rng_key, subkey = jax.random.split(rng_key)
        live, info = step(subkey, live, inner_kernel_params)
        integrator = update_integrator(integrator, live, info)
        inner_kernel_params = update_inner_kernel_params(live, info, inner_kernel_params)
        dead.append(info)
        pbar.update(len(info.particles.position))

ns_run = finalise(live, dead)
```

## Benefits

1. **Simpler state**: Flat structure, no wrapper types
2. **Explicit control**: Parameter adaptation is visible in user code
3. **Flexible adaptation**: Users can implement custom update strategies
4. **Follows patterns**: Mirrors SMC's external parameter management
5. **Better for scan**: Natural tuple packing `(state, integrator, params)`

## Trade-offs

**Advantage**: Maximal simplicity and flexibility
**Cost**: Users manage parameter updates in their outer loop (3 extra lines)

This trade-off strongly favors simplicity - the parameter update is:
- Explicit and visible
- Easy to customize
- Natural for advanced patterns (scan, while_loop)

## Migration

**Before** (parameters in state):
```python
live = algo.init(positions)
integrator = init_integrator(live)

for _ in range(n_steps):
    live, info = step(rng_key, live)
    integrator = update_integrator(integrator, live, info)
```

**After** (parameters external):
```python
live = algo.init(positions)
integrator = init_integrator(live)
inner_kernel_params = init_inner_kernel_params(live)

for _ in range(n_steps):
    live, info = step(rng_key, live, inner_kernel_params)
    integrator = update_integrator(integrator, live, info)
    inner_kernel_params = update_inner_kernel_params(live, info, inner_kernel_params)
```

### CODE REVIEW

**Comment by williamjameshandley** [comment]
Closing to reopen with correct numbering after restructuring PRs.
